packages/commonwealth/server/routing/router.ts

```diff
@@ -214,7 +216,12 @@ function setupRouter(
     updateAddress.bind(this, models)
   );
   router.get('/domain', domain.bind(this, models));
-  router.get('/status', status.bind(this, models));
+  router.get('/status', cacheDecorator.cache(60, defaultUserKeyGenerator), status.bind(this, models));
```

Comment:
Are we really sure we want to cache the results of status? What cases did we verify correctness against?

Reply:
> do we want to?
- yes, its the second most frequent route
Basically, we want to cache these findAlls

https://github.com/hicommonwealth/commonwealth/blob/e89ae04f2b2f4db8fdf8d3b431ab2e309fa57062/packages/commonwealth/server/routes/status.ts#L72
```
      models.Chain.findAll({
      models.ChainNode.findAll(),
      models.NotificationCategory.findAll(),
      models.ChainCategory.findAll(),
      models.ChainCategoryType.findAll(),
```
- is it safe? 
- Looked inside the function - no useractivity updates in status function 
- only thing seems might frequently changed is `starrred communitites`
- its done at user level right now - `route_response_96271-/api/status`

Reply:
Is redis caching of the entire route's response the right solution when we only want to cache a part of the response?

Reply:
right place to do partial caching is inside the function, added function which can help with more customized caching.
- checkCacheAndSendResponseIfFound
- checkCache
- cacheResponse

In next PR -> i will caching database query which will take of that, `findAll` responses will populate model from cache instead of database.

Reply:
So what you're saying is we've not implemented the right solution for caching in the /status route? In that case, why are we moving forward with the wrong solution? There are certainly routes where redis caching of the entire response is helpful, but my understanding from this conversation is that /status is not one of them.

Reply:
- for `/status` it will be cached per user, not as global response for every user because of user specific info
`route_response_user:1379_/api/status` - with TTL of 60secs
- when application bootstrapped it called without user as well, when user is logged, all those user will receive response from `route_response_/api/status`